docs/Developers/Linter_Bears_Advanced.rst

```diff
@@ -41,6 +40,30 @@ The path of the temporary configuration file can be accessed inside
         def create_arguments(filename, file, config_file):
             return "--use-config", config_file
 
+If you return ``None``, no configuration file is generated. A common case
+where to explicitly return ``None`` is when the user wants to use his own
+tool-specific config, to increase performance. If a config file is supplied
+by user that means he won't need the setting provided by coala to generate
+the config file. Therefore, in such scenarios, it is advised to return
+``None`` in ``generate_config()``.
+
+::
+
+    @linter(executable='...')
+    class MyBear:
+    def generate_config(filename, file, settingA):
+        if settingA:
+            config_file = ("value1 = 1\n"
+                           "value2 = 2")
+            return config_file
+        else:
+            return None
+
+    def create_arguments(filename, file, config_file, settingA):
+        if config_file:
+            return "--use-config", config file
+        return args
```

Comment:
which `args`? You should somehow use `settingA` (and it should be optional :+1:)